runtime/src/bank.rs

```diff
@@ -4043,6 +4047,19 @@ impl Bank {
         }
     }
 
+    fn apply_stake_program_fix(&mut self) {
+        if let Some(mut account) = self.get_account(&solana_stake_program::id()) {
+            self.capitalization.fetch_sub(account.lamports, Relaxed);
+            account.lamports = 0;
```

Comment:
Why do we need to drain the account here?

Reply:
Otherwise add_builtin fails due to native-program overwrite protection: https://github.com/solana-labs/solana/blob/97284adabbad84da00ae04949a9db947848135ae/runtime/src/bank.rs#L9273